File: src/services/adminService.ts
import { httpClient } from './httpClient';
import { API_ENDPOINTS } from './apiConfig';
import { StaffUser, Role, Branch, SystemConfig, Product, AuditLog, RegulatoryChartSeedResponse } from '../../types';

export interface PrivilegeLease {
    id: string;
    staffId: string;
    permission: string;
    reason: string;
    approvedBy: string;
    approvedAt: string;
    startsAt: string;
    expiresAt: string;
    isRevoked: boolean;
    isActive: boolean;
}

export interface OrassReadiness {
    profileConfigured: boolean;
    readyForSubmission: boolean;
    submissionMode: string;
    sourceReportCode: string;
    pendingReturns: number;
    returnsReadyForSubmission: number;
    missingRequirements: string[];
    notes: string[];
    lastPreparedReturnDate?: string | null;
    lastSubmissionAt?: string | null;
}

export interface OrassQueueItem {
    id: number;
    returnType: string;
    returnDate: string;
    reportingPeriodStart: string;
    reportingPeriodEnd: string;
    submissionStatus: string;
    totalRecords: number;
    isReadyForSubmission: boolean;
    validationStatus: string;
    validationMessages: string[];
}

export interface OrassSubmissionHistoryItem {
    id: number;
    returnType: string;
    returnDate: string;
    submissionStatus: string;
    submissionDate?: string | null;
    submittedBy: string;
    bogReferenceNumber: string;
    transportStatus: string;
    acknowledgementStatus: string;
    acknowledgementReference?: string | null;
    acknowledgedAt?: string | null;
    transportMessage?: string | null;
    validationMessages: string[];
}

export interface OrassSubmissionEvidence {
    returnId: number;
    transmissionId: string;
    submissionMode: string;
    endpointUrl: string;
    transportStatus: string;
    acknowledgementStatus: string;
    acknowledgementReference?: string | null;
    submittedAt?: string | null;
    acknowledgedAt?: string | null;
    payloadHash?: string | null;
    providerStatusCode?: string | null;
    transportMessage?: string | null;
    notes: string[];
}

export interface OrassReconciliationResult {
    scannedCount: number;
    updatedCount: number;
    pendingCount: number;
    executionMode: string;
    executedAt: string;
    notes: string[];
    updatedItems: OrassSubmissionHistoryItem[];
}

class AdminService {
    // Users
    async getUsers(): Promise<StaffUser[]> {
        return httpClient.get<StaffUser[]>(API_ENDPOINTS.users.list);
    }

    async createUser(data: Partial<StaffUser>): Promise<StaffUser> {
        return httpClient.post<StaffUser>(API_ENDPOINTS.users.create, data);
    }

    async updateUser(id: string, data: Partial<StaffUser>): Promise<StaffUser> {
        return httpClient.put<StaffUser>(API_ENDPOINTS.users.update(id), data);
    }

    async updateUserRole(id: string, roleId: string): Promise<StaffUser> {
        return httpClient.put<StaffUser>(API_ENDPOINTS.users.update(id), { roleId });
    }

    async deleteUser(id: string): Promise<void> {
        return httpClient.delete<void>(API_ENDPOINTS.users.delete(id));
    }

    async resetPassword(id: string): Promise<void> {
        // In a real app we'd have a specific endpoint or use update. 
        // Creating stub using PUT for now assuming backend supports password reset via update.
        return httpClient.put<void>(API_ENDPOINTS.users.update(id), { resetPassword: true, password: 'password123' });
    }

    // Roles
    async getRoles(): Promise<Role[]> {
        return httpClient.get<Role[]>(API_ENDPOINTS.users.roles);
    }

    async createRole(data: Partial<Role>): Promise<Role> {
        // Assuming POST /users/roles exists based on standard REST.
        return httpClient.post<Role>(API_ENDPOINTS.users.roles, data);
    }

    async updateRole(id: string, data: Partial<Role>): Promise<Role> {
        // Assuming PUT /users/roles/:id exists
        return httpClient.put<Role>(`${API_ENDPOINTS.users.roles}/${id}`, data);
    }

    // Branches
    async getBranches(): Promise<Branch[]> {
        return httpClient.get<Branch[]>(API_ENDPOINTS.branches.list);
    }

    async createBranch(data: Partial<Branch>): Promise<Branch> {
        return httpClient.post<Branch>(API_ENDPOINTS.branches.create, data);
    }

    async updateBranch(id: string, data: Partial<Branch>): Promise<Branch> {
        return httpClient.put<Branch>(API_ENDPOINTS.branches.update(id), data);
    }

    async deleteBranch(id: string): Promise<void> {
        // Assuming DELETE /branches/:id
        return httpClient.delete<void>(`${API_ENDPOINTS.branches.list}/${id}`);
    }

    // System Configuration
    async getSystemConfig(): Promise<SystemConfig> {
        const items = await httpClient.get<any[]>(API_ENDPOINTS.config.get);

        const config: any = {};
        const schedulerConfig = {
            enabled: false,
            timeUtc: '23:00',
            lastRunDate: '',
        };

        for (const item of items) {
            if (item.key === 'loan.credit_bureau_required_for_approval') {
                config.loanCreditBureauRequiredForApproval = item.value === 'true';
                continue;
            }

            if (item.key === 'eod_scheduler_enabled') {
                schedulerConfig.enabled = item.value === 'true';
                continue;
            }

            if (item.key === 'eod_scheduler_time_utc') {
                schedulerConfig.timeUtc = item.value || '23:00';
                continue;
            }

            if (item.key === 'eod_scheduler_last_run_date') {
                schedulerConfig.lastRunDate = item.value || '';
                continue;
            }

            try {
                // Try JSON parsing objects, strings will fallback
                config[item.key] = JSON.parse(item.value);
            } catch {
                if (!isNaN(Number(item.value))) {
                    config[item.key] = Number(item.value);
                } else if (item.value === 'true' || item.value === 'false') {
                    config[item.key] = item.value === 'true';
                } else {
                    config[item.key] = item.value;
                }
            }
        }

        config.eodScheduler = {
            ...schedulerConfig,
            ...(config.eodScheduler || {}),
        };

        return config as SystemConfig;
    }

    async updateSystemConfig(config: SystemConfig): Promise<SystemConfig> {
        // Flatten the nested UI config shape into the backend key/value store contract.
        const items = Object.entries(config)
            .filter(([key]) => key !== 'eodScheduler')
            .map(([key, value]) => ({
                key,
                value: typeof value === 'object' ? JSON.stringify(value) : String(value)
            }));

        items.push(
            { key: 'loan.credit_bureau_required_for_approval', value: String(config.loanCreditBureauRequiredForApproval) },
            { key: 'eod_scheduler_enabled', value: String(config.eodScheduler.enabled) },
            { key: 'eod_scheduler_time_utc', value: config.eodScheduler.timeUtc },
            { key: 'eod_scheduler_last_run_date', value: config.eodScheduler.lastRunDate || '' }
        );

        return httpClient.post<SystemConfig>(API_ENDPOINTS.config.update, items);
    }

    async getOrassReadiness(): Promise<OrassReadiness> {
        return httpClient.get<OrassReadiness>(API_ENDPOINTS.orass.readiness);
    }

    async getOrassQueue(): Promise<OrassQueueItem[]> {
        return httpClient.get<OrassQueueItem[]>(API_ENDPOINTS.orass.queue);
    }

    async getOrassHistory(take: number = 20): Promise<OrassSubmissionHistoryItem[]> {
        return httpClient.get<OrassSubmissionHistoryItem[]>(`${API_ENDPOINTS.orass.history}?take=${take}`);
    }

    async submitOrassReturn(returnId: number): Promise<OrassSubmissionHistoryItem> {
        return httpClient.post<OrassSubmissionHistoryItem>(API_ENDPOINTS.orass.submit(returnId), {});
    }

    async getOrassEvidence(returnId: number): Promise<OrassSubmissionEvidence> {
        return httpClient.get<OrassSubmissionEvidence>(API_ENDPOINTS.orass.evidence(returnId));
    }

    async acknowledgeOrassReturn(returnId: number, data: {
        status: 'RECEIVED' | 'ACCEPTED' | 'REJECTED';
        acknowledgementReference?: string;
        message?: string;
    }): Promise<OrassSubmissionHistoryItem> {
        return httpClient.post<OrassSubmissionHistoryItem>(API_ENDPOINTS.orass.acknowledge(returnId), data);
    }

    async reconcileOrassAcknowledgements(): Promise<OrassReconciliationResult> {
        return httpClient.post<OrassReconciliationResult>(API_ENDPOINTS.orass.reconcile, {});
    }

    async seedRegulatoryChartOfAccounts(regionCode: string = 'GH'): Promise<RegulatoryChartSeedResponse> {
        return httpClient.post<RegulatoryChartSeedResponse>(API_ENDPOINTS.gl.seedRegulatory, { regionCode });
    }

    // Privilege leases
    async getPrivilegeLeases(staffId: string): Promise<PrivilegeLease[]> {
        return httpClient.get<PrivilegeLease[]>(API_ENDPOINTS.privilegeLeases.listByStaff(staffId));
    }

    async getMyPrivilegeLeases(): Promise<PrivilegeLease[]> {
        return httpClient.get<PrivilegeLease[]>(API_ENDPOINTS.privilegeLeases.listMine);
    }

    async createPrivilegeLease(data: {
        staffId: string;
        permission: string;
        reason: string;
        approvedBy: string;
        startsAt?: string;
        expiresAt: string;
    }): Promise<PrivilegeLease> {
        return httpClient.post<PrivilegeLease>(API_ENDPOINTS.privilegeLeases.create, data);
    }

    async revokePrivilegeLease(leaseId: string, revokedBy: string): Promise<{ message: string }> {
        return httpClient.post<{ message: string }>(API_ENDPOINTS.privilegeLeases.revoke(leaseId), { revokedBy });
    }

    // Products
    async getProducts(): Promise<Product[]> {
        return httpClient.get<Product[]>(API_ENDPOINTS.products.list);
    }

    async createProduct(data: Partial<Product>): Promise<Product> {
        return httpClient.post<Product>(API_ENDPOINTS.products.create, data);
    }

    async updateProduct(id: string, data: Partial<Product>): Promise<Product> {
        return httpClient.put<Product>(API_ENDPOINTS.products.update(id), data);
    }

    // Audit Logs
    async getAuditLogs(limit: number = 100): Promise<AuditLog[]> {
        return httpClient.get<AuditLog[]>(`${API_ENDPOINTS.audit.list}?limit=${limit}`);
    }
}
export const adminService = new AdminService();

File: CoreBankerWeb/CoreBanker/Services/AdminService.cs
using System.Globalization;

namespace CoreBanker.Services
{
    public class AdminService : ApiClientBase
    {
        public AdminService(HttpClient httpClient) : base(httpClient) { }

        public async Task<List<AdminUserDto>> GetUsersAsync(CancellationToken cancellationToken = default)
        {
            var result = await GetAsync<List<AdminUserDto>>("/api/users", cancellationToken);
            return result ?? new List<AdminUserDto>();
        }

        public Task<AdminUserDto?> CreateUserAsync(CreateAdminUserRequest request, CancellationToken cancellationToken = default)
            => PostAsync<CreateAdminUserRequest, AdminUserDto>("/api/users", request, cancellationToken);

        public Task<AdminUserDto?> UpdateUserAsync(string id, UpdateAdminUserRequest request, CancellationToken cancellationToken = default)
            => PutAsync<UpdateAdminUserRequest, AdminUserDto>($"/api/users/{id}", request, cancellationToken);

        public Task DeleteUserAsync(string id, CancellationToken cancellationToken = default)
            => DeleteAsync($"/api/users/{id}", cancellationToken);

        public async Task<List<RoleSummaryDto>> GetRolesAsync(CancellationToken cancellationToken = default)
        {
            var result = await GetAsync<List<RoleSummaryDto>>("/api/roles", cancellationToken);
            return result ?? new List<RoleSummaryDto>();
        }

        public Task<RoleSummaryDto?> CreateRoleAsync(CreateRoleRequest request, CancellationToken cancellationToken = default)
            => PostAsync<CreateRoleRequest, RoleSummaryDto>("/api/roles", request, cancellationToken);

        public Task<RoleSummaryDto?> UpdateRoleAsync(string id, UpdateRoleRequest request, CancellationToken cancellationToken = default)
            => PutAsync<UpdateRoleRequest, RoleSummaryDto>($"/api/roles/{id}", request, cancellationToken);

        public async Task<List<BranchDto>> GetBranchesAsync(CancellationToken cancellationToken = default)
        {
            var result = await GetAsync<List<BranchDto>>("/api/branch", cancellationToken);
            return result ?? new List<BranchDto>();
        }

        public Task<BranchDto?> CreateBranchAsync(CreateBranchRequest request, CancellationToken cancellationToken = default)
            => PostAsync<CreateBranchRequest, BranchDto>("/api/branch", request, cancellationToken);

        public async Task UpdateBranchAsync(BranchDto request, CancellationToken cancellationToken = default)
        {
            await PutAsync<BranchDto, object>($"/api/branch/{request.Id}", request, cancellationToken);
        }

        public Task DeleteBranchAsync(string id, CancellationToken cancellationToken = default)
            => DeleteAsync($"/api/branch/{id}", cancellationToken);

        public async Task<List<PrivilegeLeaseDto>> GetPrivilegeLeasesAsync(string staffId, CancellationToken cancellationToken = default)
        {
            var result = await GetAsync<List<PrivilegeLeaseDto>>($"/api/privilege-leases/{staffId}", cancellationToken);
            return result ?? new List<PrivilegeLeaseDto>();
        }

        public Task<PrivilegeLeaseDto?> CreatePrivilegeLeaseAsync(CreatePrivilegeLeaseRequest request, CancellationToken cancellationToken = default)
            => PostAsync<CreatePrivilegeLeaseRequest, PrivilegeLeaseDto>("/api/privilege-leases", request, cancellationToken);

        public Task RevokePrivilegeLeaseAsync(string leaseId, RevokePrivilegeLeaseRequest request, CancellationToken cancellationToken = default)
            => PostAsync<RevokePrivilegeLeaseRequest, object>($"/api/privilege-leases/{leaseId}/revoke", request, cancellationToken);

        public async Task<List<ConfigItemDto>> GetConfigAsync(CancellationToken cancellationToken = default)
        {
            var result = await GetAsync<List<ConfigItemDto>>("/api/config", cancellationToken);
            return result ?? new List<ConfigItemDto>();
        }

        public async Task UpdateConfigAsync(List<ConfigItemDto> items, CancellationToken cancellationToken = default)
        {
            await PostAsync<List<ConfigItemDto>, object>("/api/config", items, cancellationToken);
        }

        public Task<OrassProfileDto?> GetOrassProfileAsync(CancellationToken cancellationToken = default)
            => GetAsync<OrassProfileDto>("/api/orass/profile", cancellationToken);

        public Task<OrassReadinessDto?> GetOrassReadinessAsync(CancellationToken cancellationToken = default)
            => GetAsync<OrassReadinessDto>("/api/orass/readiness", cancellationToken);

        public async Task<List<OrassQueueItemDto>> GetOrassQueueAsync(CancellationToken cancellationToken = default)
        {
            var result = await GetAsync<List<OrassQueueItemDto>>("/api/orass/queue", cancellationToken);
            return result ?? new List<OrassQueueItemDto>();
        }

        public async Task<List<OrassSubmissionHistoryItemDto>> GetOrassHistoryAsync(int take = 20, CancellationToken cancellationToken = default)
        {
            var result = await GetAsync<List<OrassSubmissionHistoryItemDto>>($"/api/orass/history?take={take}", cancellationToken);
            return result ?? new List<OrassSubmissionHistoryItemDto>();
        }

        public Task<OrassSubmissionHistoryItemDto?> SubmitOrassReturnAsync(int returnId, CancellationToken cancellationToken = default)
            => PostAsync<object, OrassSubmissionHistoryItemDto>($"/api/orass/submit/{returnId}", new { }, cancellationToken);

        public Task<OrassSubmissionEvidenceDto?> GetOrassEvidenceAsync(int returnId, CancellationToken cancellationToken = default)
            => GetAsync<OrassSubmissionEvidenceDto>($"/api/orass/evidence/{returnId}", cancellationToken);

        public Task<OrassSubmissionHistoryItemDto?> AcknowledgeOrassReturnAsync(int returnId, UpdateOrassAcknowledgementRequest request, CancellationToken cancellationToken = default)
            => PostAsync<UpdateOrassAcknowledgementRequest, OrassSubmissionHistoryItemDto>($"/api/orass/acknowledge/{returnId}", request, cancellationToken);

        public Task<OrassReconciliationResultDto?> ReconcileOrassAcknowledgementsAsync(CancellationToken cancellationToken = default)
            => PostAsync<object, OrassReconciliationResultDto>("/api/orass/reconcile", new { }, cancellationToken);

        public static bool GetBoolConfig(IEnumerable<ConfigItemDto> items, string key, bool defaultValue = false)
        {
            var raw = items.FirstOrDefault(item => string.Equals(item.Key, key, StringComparison.OrdinalIgnoreCase))?.Value;
            return bool.TryParse(raw, out var parsed) ? parsed : defaultValue;
        }

        public static string GetStringConfig(IEnumerable<ConfigItemDto> items, string key, string defaultValue = "")
        {
            return items.FirstOrDefault(item => string.Equals(item.Key, key, StringComparison.OrdinalIgnoreCase))?.Value ?? defaultValue;
        }

        public static decimal? GetDecimalConfig(IEnumerable<ConfigItemDto> items, string key)
        {
            var raw = items.FirstOrDefault(item => string.Equals(item.Key, key, StringComparison.OrdinalIgnoreCase))?.Value;
            return decimal.TryParse(raw, NumberStyles.Any, CultureInfo.InvariantCulture, out var parsed) ? parsed : null;
        }
    }

    public class ConfigItemDto
    {
        public string Key { get; set; } = string.Empty;
        public string Value { get; set; } = string.Empty;
    }

    public class AdminUserDto
    {
        public string Id { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string? Phone { get; set; }
        public string? BranchId { get; set; }
        public string Status { get; set; } = string.Empty;
        public string? AvatarInitials { get; set; }
        public string? AccessScopeType { get; set; }
        public DateTime? LastLogin { get; set; }
        public string? ClerkUserId { get; set; }
        public DateTime CreatedAt { get; set; }
        public List<string> RoleIds { get; set; } = new();
    }

    public class CreateAdminUserRequest
    {
        public string Name { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string? Phone { get; set; }
        public string? RoleId { get; set; }
        public string? BranchId { get; set; }
        public string? AvatarInitials { get; set; }
        public string Password { get; set; } = string.Empty;
    }

    public class UpdateAdminUserRequest
    {
        public string? Name { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string? RoleId { get; set; }
        public string? BranchId { get; set; }
        public string? Status { get; set; }
        public string? Password { get; set; }
    }

    public class RoleSummaryDto
    {
        public string Id { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public bool IsSystemRole { get; set; }
        public bool IsActive { get; set; }
        public int UserCount { get; set; }
        public List<string> Permissions { get; set; } = new();
    }

    public class CreateRoleRequest
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public List<string> Permissions { get; set; } = new();
    }

    public class UpdateRoleRequest
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public List<string> Permissions { get; set; } = new();
    }

    public class BranchDto
    {
        public string Id { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Code { get; set; } = string.Empty;
        public string? Location { get; set; }
        public string Status { get; set; } = "ACTIVE";
    }

    public class CreateBranchRequest
    {
        public string Name { get; set; } = string.Empty;
        public string Code { get; set; } = string.Empty;
        public string? Location { get; set; }
        public string Status { get; set; } = "ACTIVE";
    }

    public class CreatePrivilegeLeaseRequest
    {
        public string StaffId { get; set; } = string.Empty;
        public string Permission { get; set; } = string.Empty;
        public string Reason { get; set; } = string.Empty;
        public string ApprovedBy { get; set; } = string.Empty;
        public DateTime? StartsAt { get; set; }
        public DateTime ExpiresAt { get; set; }
    }

    public class RevokePrivilegeLeaseRequest
    {
        public string RevokedBy { get; set; } = string.Empty;
    }

    public class PrivilegeLeaseDto
    {
        public string Id { get; set; } = string.Empty;
        public string StaffId { get; set; } = string.Empty;
        public string Permission { get; set; } = string.Empty;
        public string Reason { get; set; } = string.Empty;
        public string ApprovedBy { get; set; } = string.Empty;
        public DateTime ApprovedAt { get; set; }
        public DateTime StartsAt { get; set; }
        public DateTime ExpiresAt { get; set; }
        public bool IsRevoked { get; set; }
        public bool IsActive { get; set; }
    }

    public class ConfigSnapshotDto
    {
        public bool LoanCreditBureauRequiredForApproval { get; set; }
        public bool EodSchedulerEnabled { get; set; }
        public string EodSchedulerTimeUtc { get; set; } = "23:00";
        public string EodSchedulerLastRunDate { get; set; } = string.Empty;
        public string ActiveCreditProvider { get; set; } = string.Empty;
        public decimal? DefaultTransactionLimit { get; set; }
    }

    public class OrassProfileDto
    {
        public bool Enabled { get; set; }
        public string InstitutionCode { get; set; } = string.Empty;
        public string SubmissionMode { get; set; } = "TEST";
        public string EndpointUrl { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string CertificateAlias { get; set; } = string.Empty;
        public string SourceReportCode { get; set; } = string.Empty;
        public bool AutoSubmit { get; set; }
        public string CutoffTimeUtc { get; set; } = "17:00";
        public string FallbackEmail { get; set; } = string.Empty;
        public string? LastSubmissionAt { get; set; }
    }

    public class OrassReadinessDto
    {
        public bool ProfileConfigured { get; set; }
        public bool ReadyForSubmission { get; set; }
        public string SubmissionMode { get; set; } = "TEST";
        public string SourceReportCode { get; set; } = string.Empty;
        public int PendingReturns { get; set; }
        public int ReturnsReadyForSubmission { get; set; }
        public string[] MissingRequirements { get; set; } = [];
        public string[] Notes { get; set; } = [];
        public string? LastPreparedReturnDate { get; set; }
        public string? LastSubmissionAt { get; set; }
    }

    public class OrassQueueItemDto
    {
        public int Id { get; set; }
        public string ReturnType { get; set; } = string.Empty;
        public string ReturnDate { get; set; } = string.Empty;
        public string ReportingPeriodStart { get; set; } = string.Empty;
        public string ReportingPeriodEnd { get; set; } = string.Empty;
        public string SubmissionStatus { get; set; } = string.Empty;
        public int TotalRecords { get; set; }
        public bool IsReadyForSubmission { get; set; }
        public string ValidationStatus { get; set; } = string.Empty;
        public string[] ValidationMessages { get; set; } = [];
    }

    public class OrassSubmissionHistoryItemDto
    {
        public int Id { get; set; }
        public string ReturnType { get; set; } = string.Empty;
        public string ReturnDate { get; set; } = string.Empty;
        public string SubmissionStatus { get; set; } = string.Empty;
        public string? SubmissionDate { get; set; }
        public string SubmittedBy { get; set; } = string.Empty;
        public string BogReferenceNumber { get; set; } = string.Empty;
        public string TransportStatus { get; set; } = string.Empty;
        public string AcknowledgementStatus { get; set; } = string.Empty;
        public string? AcknowledgementReference { get; set; }
        public string? AcknowledgedAt { get; set; }
        public string? TransportMessage { get; set; }
        public string[] ValidationMessages { get; set; } = [];
    }

    public class OrassSubmissionEvidenceDto
    {
        public int ReturnId { get; set; }
        public string TransmissionId { get; set; } = string.Empty;
        public string SubmissionMode { get; set; } = string.Empty;
        public string EndpointUrl { get; set; } = string.Empty;
        public string TransportStatus { get; set; } = string.Empty;
        public string AcknowledgementStatus { get; set; } = string.Empty;
        public string? AcknowledgementReference { get; set; }
        public string? SubmittedAt { get; set; }
        public string? AcknowledgedAt { get; set; }
        public string? PayloadHash { get; set; }
        public string? ProviderStatusCode { get; set; }
        public string? TransportMessage { get; set; }
        public string[] Notes { get; set; } = [];
    }

    public class OrassReconciliationResultDto
    {
        public int ScannedCount { get; set; }
        public int UpdatedCount { get; set; }
        public int PendingCount { get; set; }
        public string ExecutionMode { get; set; } = string.Empty;
        public string ExecutedAt { get; set; } = string.Empty;
        public string[] Notes { get; set; } = [];
    }

    public class UpdateOrassAcknowledgementRequest
    {
        public string Status { get; set; } = "RECEIVED";
        public string? AcknowledgementReference { get; set; }
        public string? Message { get; set; }
    }
}
